File: cslavb/Csla/Windows/CslaActionCancelEventArgs.vb
Imports System
Imports System.ComponentModel

Namespace Windows

  ''' <summary>
  ''' Event args providing information about
  '''  a canceled action.
  ''' </summary>
  ''' <remarks></remarks>
  Public Class CslaActionCancelEventArgs

    Inherits CancelEventArgs

    ''' <summary>
    ''' Creates an instance of the object.
    ''' </summary>
    ''' <param name="cancel">
    ''' Indicates whether a cancel should occur.
    ''' </param>
    ''' <param name="commandName">
    ''' Name of the command.
    ''' </param>
    Public Sub New(ByVal cancel As Boolean, ByVal commandName As String)
      MyBase.New(cancel)
      _CommandName = commandName
    End Sub

    Private _CommandName As String

    ''' <summary>
    ''' Gets the name of the command.
    ''' </summary>
    Public ReadOnly Property CommandName() As String
      Get
        Return _CommandName
      End Get
    End Property

  End Class

End Namespace

File: Source/Csla.Windows/CslaActionCancelEventArgs.cs
//-----------------------------------------------------------------------
// <copyright file="CslaActionCancelEventArgs.cs" company="Marimer LLC">
//     Copyright (c) Marimer LLC. All rights reserved.
//     Website: https://cslanet.com
// </copyright>
// <summary>Event args providing information about</summary>
//-----------------------------------------------------------------------
using System;
using System.ComponentModel;

namespace Csla.Windows
{
  /// <summary>
  /// Event args providing information about
  /// a canceled action.
  /// </summary>
  public class CslaActionCancelEventArgs : CancelEventArgs
  {
    /// <summary>
    /// Creates an instance of the object.
    /// </summary>
    /// <param name="cancel">
    /// Indicates whether a cancel should occur.
    /// </param>
    /// <param name="commandName">
    /// Name of the command.
    /// </param>
    public CslaActionCancelEventArgs(bool cancel, string commandName)
      : base(cancel)
    {
      _commandName = commandName;
    }

    private string _commandName;

    /// <summary>
    /// Gets the name of the command.
    /// </summary>
    public string CommandName
    {
      get { return _commandName; }
    }
  }
}